service/matching/taskQueueManager.go

```diff
@@ -757,7 +786,11 @@ func (mp *metadataPoller) StartIfUnstarted(pollFn func()) {
 				case <-ticker.C:
 					// In case the interval has changed
 					ticker.Reset(mp.pollIntervalCfgFn())
-					pollFn()
+					shouldDie := pollFn()
+					if shouldDie {
+						mp.launched = sync.Once{}
```

Comment:
hmm, replacing the sync.Once feels dirty. and maybe not really thread-safe? at least it's not obviously so to me.

I would probably just do this with a plain old mutex and a bool saying whether it's running or not. or I guess an atomic if you want to get fancy.

Reply:
Fixed